### PR TITLE
Adds phobias as roundstart quirks

### PR DESCRIPTION
**Pull request by Names-Are-Hard**
## About The Pull Request
Phobias are now a negative quirk option. Once phobias are taken as a quirk, you can change which phobia in the main character prefs screen.
image
image
image

## Why It's Good For The Game
Now you can have your character fear catgirls as much as you do!

## Changelog
:cl:
add: Phobias are available as quirks
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
should probably be a 0 ooint rp thing, you can just powergame and choose something that shows up really rarely

**Comment by Names-Are-Hard** [comment]
That's what I had originally, but then it showed up as a neutral quirk.

**Comment by Names-Are-Hard** [comment]
Reeeeeeee
image

File: code/modules/client/preferences.dm [hunk 1]
```diff
@@ -1394,10 +1401,15 @@ GLOBAL_LIST_EMPTY(preferences_datums)
...
 						pda_style = pickedPDAStyle
 				if("pda_color")
-					var/pickedPDAColor = input(user, "Choose your PDA Interface color.", "Character Preference",pda_color) as color|null
+					var/pickedPDAColor = input(user, "Choose your PDA Interface color.", "Character Preference", pda_color) as color|null
 					if(pickedPDAColor)
 						pda_color = pickedPDAColor
 
+				if("phobia")
+					var/phobiaType = input(user, "What are you scared of?", "Character Preference", phobia) as null|anything in SStraumas.phobia_types
+					if(phobiaType)
```
Comment (lordpidey) [inline code/modules/client/preferences.dm]: I'd attach an else here, that way they can pick again if something goes wrong.

Comment (Names-Are-Hard) [reply code/modules/client/preferences.dm]: On `if(phobiaType)`?

Comment (lordpidey) [reply code/modules/client/preferences.dm]: Yes.

Comment (Names-Are-Hard) [reply code/modules/client/preferences.dm]: They can just press the button again if it doesn't work.

**Comment by wesoda25** [comment]
Does that removal code work? Or is it curing just any trauma? 

Also it should be randomly picked from trauma pool. 

And you should try fixing the bug where you don't get trauma message at roundstart, but do at late arrival. Your code seems functionally the same as what I did in my pr (#45586) and I encountered that bug upon testing.

**Comment by Names-Are-Hard** [comment]
1. Removal code removes phobias with resilience 5, which should only be the ones added by the quirk outside of admin intervention.

2. The quirk is worth 0 points, so it can't be used to powergame more points. It exists purely as an RP thing, by letting your character fear a specific thing.

3. Bug fixing is outside the scope of what I was trying to do with this PR. I might take a look if you link me to the actual bug itself.

**Comment by wesoda25** [comment]
> 1. Removal code removes phobias with resilience 5, which should only be the ones added by the quirk outside of admin intervention.

Yes but does it cure the _specific_ trauma? If you just have it curing traumas with resilience 5, it would also cure the paraplegic trauma while ignoring the phobia. 

> 2. The quirk is worth 0 points, so it can't be used to powergame more points. It exists purely as an RP thing, by letting your character fear a specific thing.

👍 

> 3. Bug fixing is outside the scope of what I was trying to do with this PR. I might take a look if you link me to the actual bug itself.

The bug never existed because the PR was never merged. If you were to test it on a private server you could probably duplicate it yourself. As I observed it, you get the phobia add message only if you join late, but not at roundstart.

**Comment by Names-Are-Hard** [comment]
`H.cure_trauma_type(/datum/brain_trauma/mild/phobia, TRAUMA_RESILIENCE_ABSOLUTE)` It only removes phobias with resilience 5, no other traumas are affected.

**Comment by Timonkeyn** [comment]
Make rarer ones the 0 points and common ones - 1

**Comment by Names-Are-Hard** [comment]
> 
> 
> Make rarer ones the 0 points and common ones - 1

No

**Comment by Mickyan** [comment]
neutral quirks should be in neutral.dm

File: code/modules/client/preferences.dm (hunk 1 above)
Comment (lordpidey) [reply code/modules/client/preferences.dm]: But there's no indication that it didn't work, unless you're really paying attention.

**Comment by Names-Are-Hard** [comment]
I intended it to be negative, but I didn't realize that 0 point quirks would always show up as neutral.

**Comment by anconfuzedrock** [comment]
meanwhile light drinker is -1

**Comment by Names-Are-Hard** [comment]
@wesoda25 It looks like a lot of quirks don't display a message when spawning in roundstart.

**Comment by Names-Are-Hard** [comment]
In fact, it appears add() isn't called roundstart, since my character doesn't actually spawn with the trauma.

**Comment by Difrit** [comment]
This should pick a random trauma if it doesn't already, to add to the !!FUN!! value. I can't wait to be afraid of security as security.

**Comment by Names-Are-Hard** [comment]
The entire point of this PR is to have choosable phobias. If you want random phobias, you can add them in a separate PR.

**Pull request merged**